mne/realtime/tests/test_mockclient.py

```diff
@@ -184,4 +214,60 @@ def test_find_events():
         assert_true(ii == 0)
 
 
+def test_rejection():
+    event_id, tmin, tmax = 1, 0.0, 0.5
+    sfreq = 1000
+    ch_names = ['Fz', 'Cz', 'Pz', 'STI 014']
+    raw_tmax = 5
+    info = mne.create_info(ch_names=ch_names, sfreq=sfreq,
+                           ch_types=['eeg', 'eeg', 'eeg', 'stim'])
+    raw_array = np.random.randn(len(ch_names), raw_tmax * sfreq)
+    raw_array[-1, :] = 0
+    epoch_start_samples = np.arange(raw_tmax) * sfreq
+    raw_array[-1, epoch_start_samples] = event_id
+
+    reject_threshold = np.max(raw_array) - np.min(raw_array) + 1
+    reject = {'eeg': reject_threshold}
+    epochs_to_reject = [1, 3]
+    epochs_to_keep = np.setdiff1d(np.arange(len(epoch_start_samples)),
+                                  epochs_to_reject)
+    expected_drop_log = [list() for _ in range(len(epoch_start_samples))]
+    for cur_epoch in epochs_to_reject:
+        raw_array[1, epoch_start_samples[cur_epoch]] = reject_threshold + 1
+        expected_drop_log[cur_epoch] = [ch_names[1]]
+
+    raw = mne.io.RawArray(raw_array, info)
+    events = mne.find_events(raw, shortest_event=1, initial_event=True)
+    picks = mne.pick_types(raw.info, eeg=True)
+    epochs = mne.Epochs(raw, events, event_id=event_id, tmin=tmin, tmax=tmax,
+                        baseline=None, picks=picks, preload=True,
+                        reject=reject)
+    epochs_data = epochs.get_data()
+
+    assert_true(len(epochs) ==
+                len(epoch_start_samples) - len(epochs_to_reject))
+    assert_array_equal(epochs_data[:, 1, 0],
+                       raw_array[1, epoch_start_samples[epochs_to_keep]])
+    assert_array_equal(epochs.drop_log, expected_drop_log)
+    assert_array_equal(epochs.selection, epochs_to_keep)
+
+    rt_client = MockRtClient(raw)
+
+    rt_epochs = RtEpochs(rt_client, event_id, tmin, tmax, picks=picks,
+                         baseline=None, isi_max=0.5,
+                         find_events=dict(initial_event=True),
+                         reject=reject)
+
+    rt_epochs.start()
+    rt_client.send_data(rt_epochs, picks, tmin=0, tmax=raw_tmax,
+                        buffer_size=250)
+
+    assert_true(len(rt_epochs) == len(epochs_to_keep))
+    assert_array_equal(rt_epochs.drop_log, expected_drop_log)
+    assert_array_equal(rt_epochs.selection, epochs_to_keep)
+    rt_data = rt_epochs.get_data()
+    assert_true(rt_data.shape == epochs_data.shape)
+    assert_array_equal(rt_data, epochs_data)
```

Comment:
How long do these new test take relative to the ones in `master`? We have to be careful not to add too much to the testing time, I would say ideally < 1 sec for this functionality but if you need 5-10 sec to actually get something from the FT buffer we can probably live with it (but try to make this as quick as possible).